python_modules/dagster/dagster/core/definitions/asset_layer.py

```diff
@@ -294,10 +294,11 @@ def _asset_key_to_dep_node_handles(
                 )
 
     # handle internal_asset_deps
-    for node_handle, assets_defs in assets_defs_by_node_handle.items():
-        all_output_asset_keys = assets_defs.keys
-        for asset_key, dep_asset_keys in assets_defs.asset_deps.items():
-            for dep_asset_key in [key for key in dep_asset_keys if key in all_output_asset_keys]:
+    for node_handle, assets_def in assets_defs_by_node_handle.items():
+        for asset_key, dep_asset_keys in assets_def.asset_deps.items():
```

Comment:
what's going on here is that this change uncovered an unrelated bug with how we were calculating these dependency keys after doing the cycle resolution stuff.

basically, we were using the internal_asset_deps as the source of truth for which assets were produced by which node, rather that assets_def.keys.